servicetalk-dependencies/build.gradle

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright © 2018-2019 Apple Inc. and the ServiceTalk project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+apply plugin: "io.servicetalk.servicetalk-gradle-plugin-internal-core"
+apply plugin: "java-platform"
+
+description = "ServiceTalk BOM that includes direct dependencies"
+
+javaPlatform {
+  allowDependencies()
+}
+
+rootProject.subprojects.findAll {
+  !it.name.endsWith("-bom") && !it.name.endsWith("-dependencies") &&
+      !it.name.contains("examples") &&
+      !it.name.equals("grpc") && !it.name.equals("http")
+}.each {
+  dependencies.constraints.add(it.name.endsWith("-internal") ? "runtime" : "api", it)
+}
+
+dependencies {
+  api platform("io.netty:netty-bom:${nettyVersion}")
+  api platform("org.glassfish.jersey:jersey-bom:$jerseyVersion")
+
+  constraints {
+    api "com.fasterxml.jackson.core:jackson-databind:$jacksonVersion"
```

Comment:
some of the dependencies also come with a multi-module bom, e.g.,:
* jackson-bom
* protobuf-bom
* zipkin-reporter-bom
* jersey-bom

Reply:
For now I stuck with the BOMs that ServiceTalk is currently using.

Reply:
tThe misalignment issues from protobuf and jackson can be quite common though.  Constraining a single module is similar to simply having the direct dependency. It's better to import the bom for multi-module alignment.

Reply:
Follow-on PRs #2217 and #2218 now posted.